swayfmt/src/comments.rs

```diff
@@ -57,6 +57,23 @@ fn write_trailing_comment(
     Ok(())
 }
 
+/// Writes an indentation if the given `predicate` evaluates to true.
+pub fn write_indent_if(
+    predicate: bool,
```

Comment:
It seems a little odd to include an evaluated condition as the first argument: can we instead have this be `write_indent`, and then use `if` or `predicate.then(||...` to first check the condition before writing the indent?

Reply:
Actually, that was how I wrote [this](https://github.com/FuelLabs/sway/pull/4129/files#diff-43ff8fe4b8153ef64b5a5ee8c2b1cf50b5922a96de80d571ce74736ee3994153R41-R42) at first but it ended up taking a lot of vertical space:

```
 write_comments(formatted_code, end..self.value.span().start(), formatter)
                .and_then(|written|  {
                                    if written {
                                         write_indent(..)
                                    }
                                 }
```

I guess your suggested method does make it clearer :thinking: